core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -2425,6 +2449,381 @@ public static Object structAccess(Object structObject, int index, String fieldNa
     }
   }
 
+
+  private static boolean isScalarObject(Object obj) {
+    if (obj instanceof Collection) {
+      return false;
+    }
+    if (obj instanceof Map) {
+      return false;
+    }
+    return true;
+  }
+
+  public static Object jsonValueExpression(String input) {
+    try {
+      return dejsonize(input);
+    } catch (Exception e) {
+      return e;
+    }
+  }
+
+  public static Object jsonStructuredValueExpression(Object input) {
+    return input;
+  }
+
+  public static PathContext jsonApiCommonSyntax(Object input, String pathSpec) {
+    try {
+      Matcher matcher = JSON_PATH_BASE.matcher(pathSpec);
+      if (!matcher.matches()) {
+        throw new RuntimeException("illegal patch spec, missing mode declaration");
```

Comment:
Should error message start with a capital and include `pathSpec` in question?

Reply:
Agree.